src/storage/BlockCalldataStorage.ts

```diff
@@ -0,0 +1,42 @@
+import { ForkId, BlockHeight, BlockCalldata, Address } from "@/types/types";
+import { Block } from "@/models";
+
+type CalldataCoordinateKey = string;
+
+export class BlockCalldataStorage {
+    // ====================================
+    // STORAGE MAPS
+    // ====================================
+    private coordinatesToBlockMap: Map<CalldataCoordinateKey, BlockCalldata>;
+
+    constructor() {
+        this.coordinatesToBlockMap = new Map();
+    }
+
+    // ====================================
+    // CREATE
+    // ====================================
+
+    storeBlockCalldata(blockCalldata: BlockCalldata): CalldataCoordinateKey {
+        const block: Block = Block.fromSignedBlock(
+            blockCalldata.signedBlock,
+            blockCalldata.onChainTimestamp
+        );
+        const coordinateKey = `${block.forkId}:${block.height}:${block.author}`;
```

Comment:
altough only two places
but consider adding private function that converts a block into cooridantes